test/model/autocomplete_test.dart

```diff
@@ -1307,6 +1411,566 @@ void main() {
+      final result = ChannelLinkAutocompleteQuery(rawQuery).testChannel(channel, store);
+      expected
+        ? check(result).isA<ChannelLinkAutocompleteResult>()
+        : check(result).isNull();
+    }
+
+    test('channel is always excluded when archived, regardless of other criteria', () {
+      store = eg.store();
+
+      doCheck('Channel Name', eg.stream(name: 'Channel Name', isArchived: true), false);
+      // When not archived, then other criteria will be checked.
+      doCheck('Channel Name', eg.stream(name: 'Channel Name', isArchived: false), true);
+    });
+
+    test('testChannel: channel is included if name words match the query', () {
+      store = eg.store();
+
+      doCheck('', eg.stream(name: 'Channel Name'), true);
+      doCheck('', eg.stream(name: ''), true); // unlikely case, but should not crash
+      doCheck('Channel Name', eg.stream(name: 'Channel Name'), true);
+      doCheck('channel name', eg.stream(name: 'Channel Name'), true);
+      doCheck('Channel Name', eg.stream(name: 'channel name'), true);
+      doCheck('Channel', eg.stream(name: 'Channel Name'), true);
+      doCheck('Name', eg.stream(name: 'Channel Name'), true);
+      doCheck('Channel Name', eg.stream(name: 'Channels Names'), true);
+      doCheck('Channel Four', eg.stream(name: 'Channel Name Four Words'), true);
+      doCheck('Name Words', eg.stream(name: 'Channel Name Four Words'), true);
+      doCheck('Channel F', eg.stream(name: 'Channel Name Four Words'), true);
+      doCheck('C Four', eg.stream(name: 'Channel Name Four Words'), true);
+      doCheck('channel channel', eg.stream(name: 'Channel Channel Name'), true);
+      doCheck('channel channel', eg.stream(name: 'Channel Name Channel'), true);
+
+      doCheck('C', eg.stream(name: ''), false); // unlikely case, but should not crash
+      doCheck('Channels Names', eg.stream(name: 'Channel Name'), false);
+      doCheck('Channel Name', eg.stream(name: 'Channel'), false);
+      doCheck('Channel Name', eg.stream(name: 'Name'), false);
+      doCheck('nnel ame', eg.stream(name: 'Channel Name'), false);
+      doCheck('nnel Name', eg.stream(name: 'Channel Name'), false);
+      doCheck('Channel ame', eg.stream(name: 'Channel Name'), false);
+      doCheck('Channel Channel', eg.stream(name: 'Channel Name'), false);
+      doCheck('Name Name', eg.stream(name: 'Channel Name'), false);
+      doCheck('Name Channel', eg.stream(name: 'Channel Name'), false);
+      doCheck('Name Four Channel Words', eg.stream(name: 'Channel Name Four Words'), false);
+      doCheck('F Channel', eg.stream(name: 'Channel Name Four Words'), false);
+      doCheck('Four C', eg.stream(name: 'Channel Name Four Words'), false);
+    });
+
+    group('ranking', () {
+      // This gets filled here at the start of the group, but never reset.
+      // We're counting on this group's tests never doing anything to mutate it.
+      store = eg.store();
```

Comment:
This comment isn't right, because this is setting the `store` local that was declared in the outer group. That one gets overwritten in that group's other tests above.

In general it's best if the code outside of test cases (outside of `test` bodies) never creates large objects like instances of PerAccountStore. This code is declaring the tests, not part of running any specific test, so it runs unconditionally even when you use `--name` to filter down which tests run.

Hence why the "MentionAutocompleteQuery ranking" group (a) declares its own `store` local, (b) leaves it to be initialized lazily by a test case.